### ISSUE TITLE
Provide a way to programmatically save Credentials nodes (OAUTH)

### ISSUE THREAD

**Issue by stefanopog** [node-red/node-red#1055]
When a node uses OAuth to authenticate, it is likely that the AccessToken that is initially retrieved would expire in some period of time. Thus in the code you need to catch the return from an invalid token and issue the Refresh; after refreshing you need to update the Credentials in order to account for the new AccessToken and the next RefreshToken.
Since Node-Red only dumps the Credentials upon the use of the "DEPLOY" button, there is no way for the code in the node to make sure that the refreshed credentials are properly dumped. Thus as soon as Node-Red needs to restart, it is likely that a previously used set of credentials will be read, thus invalidating the node itself.

What I propose is to add a Method that "forces" the dump of the credentials, thus simulating a manual "DEPLOY".

**Comment by knolleary** [comment]
(discussed internally... raised to capture the requirement)

**Comment by stefanopog** [comment]
Any news on this ?

**Comment by knolleary** [comment]
No, this has some far reaching consequences when it comes to version control and the current api design. Will be looked at after the next release when we hope to make progress on versioning of flows.

**Comment by alexkn** [comment]
Any news on this ?
